### ISSUE TITLE
[Nexus 5x Hammerhead] Random reboots after OTA upgrade on March 28

### ISSUE THREAD

**Issue by bluuub** [copperhead/bugtracker#611]
Hi,
First of all thanks for the effort you put into this project. I joined last week and enjoyed the OS very much so far. Until the OTA update that was pushed yesterday everything worked flawlessly. Now I am having random reboots. Phone is literally shutting down and rebooting. This happens randomly (Already 4 times). 
I don't have the time to look into it right now, but I wanted to share this. Maybe this is not related to my installation but a general bug in the Mar 28 Kernel. Maybe this is related to #610? I opend another issue just in case it wasn't.

**Comment by thestinger** [comment]
You'll either need to provide logs via a debug cable or reproducible instructions to trigger this issue, or it can't be acted upon or treated as an issue.

**Comment by losuler** [comment]
I can confirm this is also occurring on another Nexus5X (N4F26T.2017.03.28.23.38.21). Happened twice to me, both times while interacting inside an app (Fennec F-Droid while opening a site then with K-9 Mail when opening an e-mail).

It seemingly occurs randomly, so I'm not sure how to reproduce this exactly, but I'll post logs if I manage to capture them.

**Comment by thestinger** [comment]
You probably won't be able to capture logs via adb, only a serial debug cable. Reproducible instructions for triggering it will be needed if you don't have one.

**Comment by thestinger** [comment]
Need to know the specific model of Nexus 5X that's being used, i.e. whether it's North America or International and 32GB vs. 128GB.

**Comment by thestinger** [comment]
Haven't been able to reproduce this on a 32GB North America model.

**Comment by thestinger** [comment]
Also need to know when this happens. Does it happen when you're not using the device? Does networking trigger it? WiFi? Cellular data? etc.

**Comment by losuler** [comment]
Damn, I wasn't aware of the need of such a cable. I briefly looked this up and only saw mention of DIY makes of serial cables for the Nexus 4 and 5. Is this the same situation for the 5X?

Nonetheless, I'll continue to see if I can consistently reproduce a crash. It's only occurred during use for me and both very soon after each other (10 minutes or so) and not since (30 minutes now).

I'm on the 32GB international version and so far this has occurred on cellular data (though I haven't been able to verify if this also occurs on WiFi yet).

**Comment by bluuub** [comment]
International Model (EU), 16 GB
I don't own a serial cable and have not experience in using one. In my case the random reboot always occured when I was doing something, i.e. opening an App, etc.. It did not happen while the device was idle (at least not until now). 
I don't know the process, but can you provide an option to downgrade to the version before (that one had the March security updates as well so using that one for now would be no security issue) over OTA? That would be a real great short term solution if it does not put too much work on your shoulders. Because flashing manually will erase the data right?

**Comment by thestinger** [comment]
No, I can't provide the option to downgrade and there will be a security update on Monday so you'll need to be upgraded for that. The solution to this is determining what is going wrong and fixing the cause, not downgrading.

**Comment by thestinger** [comment]
So the common thing here that's different from our devices is that you're both on International models.

**Comment by ScottyBauer** [comment]
Hi,
Can you please attach the phone and enable usb debugging. From there go adb shell. After that send us the output of `cat /sys/fs/pstore/console-ramoops`

**Comment by thestinger** [comment]
It seems very likely that this is caused by WiFi, and strangely it seems specific to Europe.

**Comment by losuler** [comment]
This doesn't seem to be consistent with my narrative however. I'm from Australia and it occurred to me twice on cellular data. I was travelling at the time that the crashes occurred and hadn't since I stopped. I'll reproduce those conditions and see if it causes it to crash again.

**Comment by ScottyBauer** [comment]
@losuler  Yeah, it should stick around(the crash log), assuming you don't reboot, even if you do it will still probably be there.

**Comment by thestinger** [comment]
This likely doesn't happen for the latest release. It's not fixed, but rather the hardening uncovering the issue was removed for the time being.

**Comment by thestinger** [comment]
Open a new issue if anything still happens, because these were likely not all the same issue anyway.
